components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/utils/ContentSearchResultNameComparator.java

```diff
@@ -27,29 +28,101 @@
+        } else if (o1 instanceof APIProduct && o2 instanceof APIProduct) {
+            APIProduct apiProduct1 = (APIProduct) o1;
+            APIProduct apiProduct2 = (APIProduct) o2;
+            return productNameComparator.compare(apiProduct1, apiProduct2);
+        } else if (o1 instanceof API && o2 instanceof APIProduct) {
+            API api = (API) o1;
+            APIProduct apiProduct = (APIProduct) o2;
+            return api.getId().getApiName().compareToIgnoreCase(apiProduct.getId().getName());
+        } else if (o1 instanceof APIProduct && o2 instanceof API) {
+            APIProduct apiProduct = (APIProduct) o1;
+            API api = (API) o2;
+            return apiProduct.getId().getName().compareToIgnoreCase(api.getId().getName());
         } else if (o1 instanceof API && o2 instanceof Map.Entry) {
             API api = (API) o1;
-            Map.Entry<Documentation, API> documentEntry = (Map.Entry<Documentation, API>) o2;
-            Documentation doc = documentEntry.getKey();
-            return api.getId().getApiName().compareToIgnoreCase(doc.getName());
+            if (((Map.Entry) o2).getValue() instanceof API) {
+                Map.Entry<Documentation, API> documentEntry = (Map.Entry<Documentation, API>) o2;
+                Documentation doc = documentEntry.getKey();
+                return api.getId().getApiName().compareToIgnoreCase(doc.getName());
+            } else {
+                Map.Entry<Documentation, APIProduct> documentEntry = (Map.Entry<Documentation, APIProduct>) o2;
+                Documentation doc = documentEntry.getKey();
+                return api.getId().getApiName().compareToIgnoreCase(doc.getName());
+            }
         } else if (o1 instanceof Map.Entry && o2 instanceof API) {
-            Map.Entry<Documentation, API> documentEntry = (Map.Entry<Documentation, API>) o1;
-            Documentation doc = documentEntry.getKey();
             API api = (API) o2;
-            return doc.getName().compareToIgnoreCase(api.getId().getApiName());
-        } else{
-            Map.Entry<Documentation, API> documentEntry1 = (Map.Entry<Documentation, API>) o1;
-            Documentation doc1 = documentEntry1.getKey();
-            Map.Entry<Documentation, API> documentEntry2 = (Map.Entry<Documentation, API>) o2;
-            Documentation doc2 = documentEntry2.getKey();
+            if (((Map.Entry) o1).getValue() instanceof API) {
+                Map.Entry<Documentation, API> documentEntry = (Map.Entry<Documentation, API>) o1;
+                Documentation doc = documentEntry.getKey();
+                return api.getId().getApiName().compareToIgnoreCase(doc.getName());
+            } else {
+                Map.Entry<Documentation, APIProduct> documentEntry = (Map.Entry<Documentation, APIProduct>) o1;
+                Documentation doc = documentEntry.getKey();
+                return api.getId().getApiName().compareToIgnoreCase(doc.getName());
```

Comment:
Difference between line 57 to 65 and line 68 to 75 is o1 and o2
Isn't it better to move the snippet to a method and reuse to avoid code duplication.